tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/PrehashedSortedMap.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.registry;
+
+import com.google.common.collect.ForwardingSortedMap;
+import java.util.SortedMap;
+
+/**
+ * A sorted map implementation which prehashes for faster usage in hashmaps.
+ * This is only safe for immutable underlying maps.
```

Comment:
this assumes not only immutable map (i.e. can't add/remove from map), but also immutable entries such that you can't change the value for a given key, and that the keys/values themselves can't mutate or else the hashcode is no longer valid, and bad things happen

Reply:
yep - changed comment